CONTRIBUTING.md

```diff
@@ -1,36 +1,202 @@
 # Contributing to Noir
 
-Thank you for your interest in contributing to Noir! There are many ways to contribute and we appreciate all of them.
+Thank you for your interest in contributing to Noir! We value your contributions in making Noir better.
 
-## Asking Questions
+This guide will discuss how the Noir team handles [Commits](#commits), [Pull Requests](#pull-requests), [Merging](#merging), [Releases](#releases), and the [Changelog](#changelog).
 
-The best place to ask general questions about Noir is in the #noir channel on the [Aztec discord](https://discord.gg/aztec). 
+__Note:__ We won't force external contributors to follow this verbatim, but following these guidelines definitely helps us in accepting your contributions.
 
-It might also be helpful to review existing and previous Github issues to see if your question has been answered already. 
+## Commits
 
-For specific technical questions, submitting an issue in this repo may be the best place to ask. 
+We want to keep our commits small and focused. This allows for easily reviewing individual commits and/or splitting up pull requests when they grow too big. Additionally, this allows us to merge smaller changes quicker and release more often.
 
-## Giving Feedback
+When committing, it's often useful to use the `git add -p` workflow to decide on what parts of the changeset to stage for commit. When making the commit, write the commit message as a Conventional Commit.
 
-Feedback is a gift. The #noir channel on Discord is a great synchronous channel for feedback. Reporting issues that you encounter is also a great channel for sharing feedback.
+### Conventional Commits
 
-## Submitting Issues
+As of the [1234567](https://github.com/noir-lang/noir/commit/1234567) commit, Noir follows the [Conventional Commits (v1.0.0)](https://www.conventionalcommits.org/en/v1.0.0/) specification. Following this convention allows us to provide an automated release process that also generates a detailed Changelog.
```

Comment:
Just making a note that we need to remember to come back and update this commit later.

Reply:
Let's call it d78c938cb260368bcf5c7a54ced9ae83bfc3d800
```suggestion
As of the [d78c938](https://github.com/noir-lang/noir/commit/d78c938cb260368bcf5c7a54ced9ae83bfc3d800) commit, Noir follows the [Conventional Commits (v1.0.0)](https://www.conventionalcommits.org/en/v1.0.0/) specification. Following this convention allows us to provide an automated release process that also generates a detailed Changelog.
```